### ISSUE TITLE
Etiketten mit erstem Schlagwort/Genre

### ISSUE THREAD

**Issue by fb4thxUall** [jzakotnik/openlibry#142]
Ich versuche mich mal selbst und mache dann einen PR, falls erfolgreich.

**Comment by jzakotnik** [comment]
Viel Erfolg, sag wenn ich was helfen kann. Allerdings obacht dass wir nicht die gleiche Datei editieren und in ein merge-hell reinlaufen. Für das Filtern der Etiketten bin ich auch grade an `pages/api/report/booklabels/index.tsx` dran im PR #140

**Comment by fb4thxUall** [comment]
Ja, genau an der Datei arbeite ich natürlich auch :-D. Ich warte aber, bis deine Veränderungen drin sind und füge meine erst anschließend in die dann aktuelle Datei ein.


Habs sogar "schon" fertig (wie das halt ist, wenn man sich in eine neue Sprache reinfuchsen muss :-D), funktioniert wie ich mir das vorstelle. 
Hab die Struktur der Datei dabei ehrlich gesagt noch nicht richtig verstanden, mir nur das Nötigste erschlossen, um es an die richtige Position zu fummeln. Aber irgendwo fängt man halt an. Sobald du deins hast, füge ich meins dazu.

**Comment by fb4thxUall** [comment]
Hab es schonmal in mein Repo geladen, damit du es schon ansehen kannst, falls du das möchtest.

Verbesserungsvorschläge natürlich sehr gerne und wenn es es ablehnen möchtest, ist auch alles gut. Freue mich über Korrekturen/Verbesserungsinput.

**Comment by jzakotnik** [comment]
Ja super! Kannst Du direkt ein PR erzeugen, dann kann man besser reinkommentieren und feedback geben

**Comment by jzakotnik** [comment]
Kurze Frage, was findest Du besser - an sich funktioniert es und wir können es so in die Codebase aufnehmen. Oder lieber die Learning Experience wie man manche Sachen in js löst und dann noch ändern? Ist beides ok für mich, ich finds ja schon mega, dass man nicht mehr allein rumwerkelt 🌻

**Comment by fb4thxUall** [comment]
Nene, schon richtig lernen. Das ist ja Sinn der Sache. Pfusch als Einstieg in die richtige Lösung :-D

Ich will das grundsätzlich besser verstehen, auch, damit ich schneller und besser entwickeln kann. Ich hatte mit Javascript und TS bisher nie zu tun und verstehe die Konzepte noch nicht. 

Am 30. August 2024 19:29:44 MESZ schrieb Jure Zakotnik ***@***.***>:
>Kurze Frage, was findest Du besser - an sich funktioniert es und wir können es so in die Codebase aufnehmen. Oder lieber die Learning Experience wie man manche Sachen in js löst und dann noch ändern? Ist beides ok für mich, ich finds ja schon mega, dass man nicht mehr allein rumwerkelt 🌻 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/jzakotnik/openlibry/issues/142#issuecomment-2322029722
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by jzakotnik** [comment]
Ja gar kein Problem, ich wollte nur nicht, dass es als Kritik oder so aufgefasst wird, so ist es nicht gemeint. Pfusch ist es auch nicht, nur ist javascript recht funktional, also als Daumenregel braucht man in 99% der Fälle weder `var` bzw. `let` mit re-assignment der Variable noch braucht man `for` Schleifen. Bin auch noch nicht so lang dabei, aber wenn man den Bogen raus hat, dann macht's richtig Spaß!

**Comment by fb4thxUall** [comment]
Angesichts meiner Alltags-Realitäten seit diesem Schuljahr muss ich leider einsehen, dass ich hier im Moment nicht selber weiter implementieren kann. Tut mir leid. Vielleicht wird das wieder anders, aber im Moment muss ich mich auf Testen und Feedback beschränken. Alles andere wäre gerade unehrlich.

**Comment by jzakotnik** [comment]
Gar kein Ding, bischen ähnliche Situation hier. Aber es wird auch wieder Zeiten geben wo mehr Muße ist, gut Ding will Weile haben 🐯

**Comment by fb4thxUall** [comment]
Da das Feature realisiert ist und ich gerade keinen offenen Punkt mehr sehe: closed
